test/integration/pipeline/multisig_tx_pipeline_test.cpp

```diff
@@ -120,3 +150,91 @@ TEST_F(MstPipelineTest, OnePeerSendsTest) {
         ASSERT_EQ(proposal->transactions().size(), 1);
       });
 }
+
+/**
+ * @given a user that has sent a semi-signed transaction to a ledger
+ * @when the user requests pending transactions
+ * @then user's semi-signed transaction is returned
+ */
+TEST_F(MstPipelineTest, GetPendingTxsAwaitingForThisPeer) {
+  auto pending_tx = baseTx()
+                        .setAccountDetail(kUserId, "fav_meme", "doge")
+                        .quorum(kSignatories + 1);
+
+  auto &mst_itf = prepareMstItf();
+  auto signed_tx = signTx(pending_tx, kUserKeypair);
+
+  auto pending_tx_check = [pending_hash = signed_tx.hash()](auto &response) {
+    ASSERT_NO_THROW({
+      const auto &pending_tx_resp = boost::apply_visitor(
+          framework::SpecifiedVisitor<
+              shared_model::interface::TransactionsResponse>(),
+          response.get());
+      ASSERT_EQ(pending_tx_resp.transactions().front().hash(), pending_hash);
+    });
+  };
+
+  // send pending transaction, signing it only with one signatory
+  mst_itf.sendTx(signed_tx).sendQuery(
+      makeGetPendingTxsQuery(kUserId, kUserKeypair), pending_tx_check);
+}
+
+/**
+ * @given a ledger with pending transactions, which lack two or more signatures
+ * @when signing those transactions with one signature @and executing get
+ * pending transactions
+ * @then they are returned with initial number of signatures plus one
+ */
+TEST_F(MstPipelineTest, GetPendingTxsLatestSignatures) {
+  auto pending_tx = baseTx()
+                        .setAccountDetail(kUserId, "fav_meme", "doge")
+                        .quorum(kSignatories + 1);
+  auto signatory_check = [](size_t expected_signatures_number) {
+    return [expected_signatures_number](auto &response) {
+      ASSERT_NO_THROW({
+        const auto &pending_tx_resp = boost::apply_visitor(
+            framework::SpecifiedVisitor<
+                shared_model::interface::TransactionsResponse>(),
+            response.get());
+        ASSERT_EQ(
+            boost::size(pending_tx_resp.transactions().front().signatures()),
+            expected_signatures_number);
+      });
+    };
+  };
+
+  prepareMstItf()
+      .sendTx(signTx(pending_tx, signatories[0]))
+      .sendQuery(makeGetPendingTxsQuery(kUserId, kUserKeypair),
+                 signatory_check(1))
+      .sendTx(signTx(pending_tx, signatories[1]))
+      .sendQuery(makeGetPendingTxsQuery(kUserId, kUserKeypair),
+                 signatory_check(2));
+}
+
+/**
+ * @given a ledger with pending transactions
```

Comment:
Seems that ledger initially empty or at least doesn't contains pending transactions.